testutils/src/main/java/com/amplifyframework/testutils/sync/SynchronousDataStore.java

```diff
@@ -33,20 +34,21 @@
  * that a DataStore operation has completed with some kind of terminal result.
  */
 public final class SynchronousDataStore {
-    private static SynchronousDataStore singleton;
+    private final DataStoreCategoryBehavior asyncDelegate;
 
-    private SynchronousDataStore() {}
+    private SynchronousDataStore(DataStoreCategoryBehavior asyncDelegate) {
+        this.asyncDelegate = asyncDelegate;
+    }
 
     /**
-     * Gets a singleton instance of the SynchronousDataStore.
-     * @return Singleton instance of Synchronous DataStore
+     * Creates a synchronizing wrapper around the provided {@link DataStoreCategoryBehavior}.
+     * @param asyncDelegate This performs actual DataStore operations
+     * @return A synchronizing wrapper around the provided {@link DataStoreCategoryBehavior}
      */
     @NonNull
-    public static synchronized SynchronousDataStore singleton() {
-        if (SynchronousDataStore.singleton == null) {
-            SynchronousDataStore.singleton = new SynchronousDataStore();
-        }
-        return SynchronousDataStore.singleton;
+    public static SynchronousDataStore delegatingTo(@NonNull DataStoreCategoryBehavior asyncDelegate) {
```

Comment:
is this supposed to change to `delegateTo`?

Reply:
Thanks for pointing this out! I have been inconsistent so far between the two in the various categories. Right now in `master`, `SynchronousApi` and `SynchronousStorage` use `delegatingTo`, so I guess I'll try to stick with that here, as well.